### DISCUSSION TITLE
Chez Scheme on ARM64

### DISCUSSION THREAD

**Discussion by michaelmesser** [Homebrew/discussions#3195] [Tap maintenance and Homebrew development]
Chez scheme does not currently support Apple Silicon. There is a [PR](https://github.com/cisco/ChezScheme/pull/607), but it is not complete yet and there is no set timeline for merging it. Is it possible for homebrew to use [Racket's fork of Chez](https://github.com/racket/ChezScheme), which does support ARM64?

**Comment by SMillerDev** [comment]
We'll only use official sources, however if the PR can be made into a patch that could be applied to the current formula.

**Reply by michaelmesser** [reply]
Could the Racket's fork of Chez be made into a patch that could be applied to the current formula? I do not believe the Chez PR is finished and there hasn't been progress for some time. Or could a separate formula be made?

**Reply by SMillerDev** [reply]
It's really much better for everyone to try and help the upstream pull request since otherwise patches and fixes would be scattered all over the internet.

**Reply by michaelmesser** [reply]
Unfortunately, I don't think many people have the skills to implement that. I thinking making a new formula or adding some kind of variant to the chez formula would be easiest.

**Reply by gwerbin** [reply]
What about a Tap? `brew install racket/racket/chez-scheme` or similar. If the forked build process is similar to or the same as the upstream build process, it should be as easy as copy-pasting the formula, and updating the URL and hashes.

**Comment by ashton314** [comment]
The Racket fork of Chez scheme has been merged into the original Chez branch; I believe this means Chez should run on Apple Silicon now.
